internal/dag/gatwwayapi_processor.go

```diff
@@ -87,7 +87,7 @@ func (p *ServiceAPIsProcessor) computeHTTPRoute(route *serviceapis.HTTPRoute) {
 			meta := types.NamespacedName{Name: *forward.ServiceName, Namespace: route.Namespace}
 
 			// TODO: Refactor EnsureService to take an int32 so conversion to intstr is not needed.
-			service, err := p.dag.EnsureService(meta, intstr.FromInt(int(forward.Port)), p.source)
+			service, err := p.dag.EnsureService(meta, intstr.FromInt(int(*forward.Port)), p.source)
```

Comment:
this will panic if `forward.Port` is nil, we should check it (not sure what the expected behavior is if it is nil, though)

Reply:
I did a check for this and also refactored a bit of logic to allow us to test this.